interactors/src/test/kotlin/interactors/StudentsInteractorTest.kt

```diff
@@ -0,0 +1,60 @@
+package com.students.results.interactors
+
+import arrow.core.Either
+import com.students.results.entities.Exam
+import com.students.results.entities.Student
+import com.students.results.repository.ExamsRepository
+import com.students.results.repository.StudentsRepository
+import com.students.results.services.requests.NotateExam
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import org.amshove.kluent.`should be`
+import org.amshove.kluent.shouldBe
+import org.amshove.kluent.shouldEqual
+import org.jetbrains.spek.api.Spek
+import org.jetbrains.spek.api.dsl.given
+import org.jetbrains.spek.api.dsl.it
+
+
+class StudentsInteractorTest : Spek({
+
+    val student = Student(id = 10L)
+    given("a student interactor") {
+
+        val studentSlot = slot<Student>()
+        val studentRepository = mockk<StudentsRepository>() {
+            every { findStudentById(1) } returns Either.right(student)
+            every { save(student = capture(studentSlot)) } returns Either.right(Unit)
+        }
+        val exam = Exam(id = 5)
+        val examsRepository = mockk<ExamsRepository>().apply {
+            every { findExamById(5) } returns Either.right(exam)
+        }
+        val studentsInteractor = StudentsInteractor(studentRepository, examsRepository)
+        studentsInteractor.notate20().apply {
+            it("should notate an existing exam with a note of 20") {
+                isRight() `should be` true
```

Comment:
The actual grade isn't checked here. How do we know it's 20?

Reply:
Yes this test should be deleted because check is done in the last test